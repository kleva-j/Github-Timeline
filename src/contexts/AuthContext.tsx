import React, { createContext, useContext, useReducer } from 'react';

import { AuthReducer, initialState, IState } from 'reducers/auth';

export const AuthStateContext = createContext<IState>(initialState);
export const AuthDispatchContext = createContext<React.Dispatch<any> | null>(null);

export const useAuthState = () => {
  const context = useContext(AuthStateContext);
  // console.log(context, 'Check here')
  if (context === undefined) {
    throw new Error('useAuthState must be used within a AuthProvider');
  }
  return context;
};

export const useAuthDispatch = () => {
  const context = useContext(AuthDispatchContext);
  if (context === undefined) {
    throw new Error('useAuthDispatch must be used within a AuthProvider');
  }
  return context;
};

export const AuthProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [user, dispatch] = useReducer(AuthReducer, initialState);

  return (
    <AuthStateContext.Provider value={user}>
      <AuthDispatchContext.Provider value={dispatch}>{children}</AuthDispatchContext.Provider>
    </AuthStateContext.Provider>
  );
};
