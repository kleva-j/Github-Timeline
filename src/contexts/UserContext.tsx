import React, { createContext, useReducer, useContext } from 'react';

import { initialState, UserReducer } from 'reducers/user';

export interface UserInterface {
  currentUser: User | undefined;
  loading: boolean;
  isAuthenticated: boolean;
  error: Error | null;
  theme?: any;
}

type User = {
  username: string | undefined;
};

export const UserStateContext = createContext<UserInterface>(initialState);
export const UserDispatchContext = createContext<React.Dispatch<any> | null>(null);

export const useUserState = () => {
  const context = useContext(UserStateContext);
  if (context === undefined) {
    throw new Error('useUserState must be used within a UserProvider');
  }
  return context;
};

export const useUserDispatch = () => {
  const context = useContext(UserDispatchContext);
  if (context === undefined) {
    throw new Error('useUserDispatch must be used within a UserProvider');
  }
  return context;
};

export const UserProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [user, dispatch] = useReducer(UserReducer, initialState);

  return (
    <UserStateContext.Provider value={user}>
      <UserDispatchContext.Provider value={dispatch}>{children}</UserDispatchContext.Provider>
    </UserStateContext.Provider>
  );
};
