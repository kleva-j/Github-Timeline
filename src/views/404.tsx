export const NotFound = () => (
  <div className="bg-gray-800 h-screen">
    <div className="flex items-center justify-center py-12">
      <div className="bg-white border rounded-md flex items-center justify-center mx-4 md:w-2/3 ">
        <div className="flex flex-col items-center py-16 ">
          <img
            className="px-4 hidden md:block"
            src="https://i.ibb.co/9Vs73RF/undraw-page-not-found-su7k-1-3.png"
            alt=""
          />
          <img className="md:hidden" src="https://i.ibb.co/RgYQvV7/undraw-page-not-found-su7k-1.png" alt="" />
          <h1 className="px-4 pt-8 pb-4 text-center text-5xl font-bold leading-10 text-gray-800">OOPS! </h1>
          <p className="px-4 pb-10 text-base leading-none text-center text-gray-600">
            No signal here! we cannot find the page you are looking for{' '}
          </p>
          <button className="mx-4 h-10 w-44 border rounded-md text-white text-base bg-indigo-700 hover:bg-indigo-800 focus:outline-none focus:ring-2 focus:ring-opacity-50 focus:ring-indigo-800">
            Go Back
          </button>
        </div>
      </div>
    </div>
  </div>
);
