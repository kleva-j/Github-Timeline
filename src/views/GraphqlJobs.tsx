import { FC } from 'react';
import { Helmet } from 'react-helmet';

import { Navbar } from 'components/Navbar';
import { GetAllJobs } from 'components/GraphqlJobs/AllJobs';

export const GraphQLJobs: FC = () => {
  return (
    <>
      <Helmet>
        <title>GraphQL Jobs - Github Timeline</title>
        <meta name="description" content="Gitline jobs page" />
      </Helmet>
      <section className="min-h-screen">
        <Navbar transparent={false} />
        <article className="my-4 px-3">
          <GetAllJobs />
        </article>
      </section>
    </>
  );
};
