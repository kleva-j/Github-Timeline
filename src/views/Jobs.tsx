import React from 'react';
import { useQuery } from '@apollo/client';

import { Footer } from 'components/Footer';
import { Navbar } from 'components/Navbar';
import { TagList } from 'components/TagList';
import { Pager } from 'components/Pagination';
import { SearchForms } from 'components/Form';
import { GetAllJobs } from 'components/AllJobs';

import { LOAD_JOBS } from 'GraphQL/Queries';

export const Jobs: React.FC = () => {
  const { loading } = useQuery(LOAD_JOBS);

  return (
    <section className="min-h-screen">
      <Navbar transparent={false} />
      <article className="my-4 px-3">
        <SearchForms />
        <TagList />
        <GetAllJobs />
        <Pager />
      </article>
      <Footer absolute={loading} />
    </section>
  );
};
