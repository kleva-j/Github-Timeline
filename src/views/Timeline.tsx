import React from 'react';

import { Footer } from 'components/Footer';
import { Navbar } from 'components/Navbar';

export const Timeline: React.FC = () => (
  <>
    <Navbar transparent={false} />
    <article className="my-10">
      <h2 className="text-gray-500 text-center">This is the Timeline page.</h2>
    </article>
    <Footer absolute />
  </>
);
