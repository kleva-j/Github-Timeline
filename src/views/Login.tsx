import { Helmet } from 'react-helmet';
import React, { useEffect } from 'react';
import { useHistory, Route } from 'react-router-dom';

import { Button, LoginBg } from 'styled';

import { Footer } from 'components/Footer';
import { Navbar } from 'components/Navbar';

import { useAuthState } from 'contexts/AuthContext';

import githubImg from 'assets/img/github.svg';
import bgImg from 'assets/img/bg_1.png';

export const Login: React.FC = () => {
  let history = useHistory();
  const context = useAuthState();
  const { isAuthenticated } = context;

  useEffect(() => {
    isAuthenticated && history.push('/timeline');
  }, [history, isAuthenticated]);

  return (
    <Route exact path="/login">
      <Helmet>
        <title>Login - Github Timeline</title>
        <meta name="description" content="Gitline login page" />
      </Helmet>
      <Navbar transparent />
      <main>
        <section className="absolute w-full h-full">
          <LoginBg url={bgImg} />

          <div className="container mx-auto px-4 h-full">
            <div className="flex content-center items-center justify-center h-full">
              <div className="w-2/3 md:w-4/12 lg:w-3/12 xl:w-2/12 px-4 mb-12">
                <div className="relative flex flex-col min-w-0 break-words w-full shadow-lg rounded-lg bg-gray-300 border-0">
                  <div className="rounded-t mb-0 py-6">
                    <div className="text-center mb-3">
                      <h6 className="text-gray-600 text-md font-bold">Sign in with</h6>
                    </div>
                    <div className="btn-wrapper text-center">
                      <a href={process.env.REACT_APP_LOGIN_ENDPOINT} target="_self" rel="noopener noreferrer">
                        <Button>
                          <img alt="Github logo" className="w-5 mr-1" src={githubImg} />
                          Github
                        </Button>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>

          <Footer absolute />
        </section>
      </main>
    </Route>
  );
};
