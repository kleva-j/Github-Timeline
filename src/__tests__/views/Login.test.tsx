import { MemoryRouter } from 'react-router-dom';
import { cleanup, render } from '@testing-library/react';

import { Login } from 'views/Login';
import { AuthProvider } from 'contexts/AuthContext';

describe('Login Page', () => {
  afterEach(cleanup);

  test('should render the Login view', () => {
    const { container, getByText } = render(
      <AuthProvider>
        <MemoryRouter initialEntries={['/login']}>
          <Login />
        </MemoryRouter>
      </AuthProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText('Sign in with')).toBeTruthy();
  });
});
