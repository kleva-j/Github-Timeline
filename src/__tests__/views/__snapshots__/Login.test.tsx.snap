// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render the Login view 1`] = `
<nav
  class="top-0 absolute z-50 w-full flex flex-wrap items-center justify-between px-2 py-3 navbar-expand-lg"
  data-testid="navbar"
>
  <div
    class="container px-4 mx-auto flex flex-wrap items-center justify-between"
  >
    <div
      class="w-full relative flex justify-between lg:w-auto lg:static lg:block lg:justify-start"
    >
      <a
        class="text-white text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-no-wrap uppercase"
        href="/"
      >
        Github Timeline
      </a>
      <button
        class="cursor-pointer text-xl leading-none px-3 py-1 border border-solid border-transparent rounded bg-transparent block lg:hidden outline-none focus:outline-none"
        type="button"
      >
        <i
          class="text-white fas fa-bars"
        />
      </button>
    </div>
    <div
      class="lg:flex flex-grow items-center bg-white lg:bg-transparent lg:shadow-none hidden"
      data-testid="list"
      id="example-navbar-warning"
    >
      <ul
        class="flex flex-col lg:flex-row list-none mr-auto"
      >
        <li
          class="flex items-center"
        >
          <a
            class="lg:text-white lg:hover:text-gray-300 text-gray-800 px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold"
            href="/jobs"
          >
            <i
              class="lg:text-gray-300 text-gray-500 far fa-file-alt text-lg leading-lg mr-2"
            />
             
            Jobs
          </a>
        </li>
      </ul>
      <ul
        class="flex flex-col lg:flex-row list-none lg:ml-auto"
      >
        <li
          class="flex items-center"
        >
          <a
            class="lg:text-white lg:hover:text-gray-300 text-gray-800 px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold"
            href="/login#twitter"
          >
            <i
              class="lg:text-gray-300 text-gray-500 fab fa-twitter text-lg leading-lg "
            />
            <span
              class="lg:hidden inline-block ml-2"
            >
              Tweet
            </span>
          </a>
        </li>
        <li
          class="flex items-center"
        >
          <a
            class="lg:text-white lg:hover:text-gray-300 text-gray-800 px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold"
            href="/login#github-stars"
          >
            <i
              class="lg:text-gray-300 text-gray-500 fab fa-github text-lg leading-lg "
            />
            <span
              class="lg:hidden inline-block ml-2"
            >
              Star
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
